virttest/utils_misc.py

```diff
@@ -3987,3 +3989,34 @@ def setup_remote(self):
         logging.debug("To check remote boolean value: %s", boolean_curr)
         if boolean_curr != self.remote_bool_value:
             raise exceptions.TestFail(result.stderr.strip())
+
+
+def get_model_features(model_name):
+    """
+    /usr/share/libvirt/cpu_map.xml defines all CPU models.
+    One CPU model is a set of features.
+    This function is to get features of one specific model.
+
+    :params model_name: CPU model name, valid name is given in cpu_map.xml
+    :return: feature list, like ['apic', 'ss']
+
+    """
+    features = []
+    conf = "/usr/share/libvirt/cpu_map.xml"
+    if not os.path.isfile(conf):
+        logging.warn("Configuration file is not found or not file: %s" % conf)
+        return []
+
+    output = process.run("cat %s" % conf)
```

Comment:
suggest use with ... statement since it help you handle connection exception gracefully